lib/streamlit/watcher/LocalSourcesWatcher.py

```diff
@@ -94,10 +94,21 @@ def on_file_changed(self, filepath):
             LOGGER.error("Received event for non-watched file", filepath)
             return
 
-        wm = self._watched_modules[filepath]
-
-        if wm.module_name is not None and wm.module_name in sys.modules:
-            del sys.modules[wm.module_name]
+        # Workaround:
+        # Delete all imported modules so we can guarantee changes to the
+        # updated module are reflected on reload.
```

Comment:
Should be "Delete all **watched** modules"

Reply:
Done 😄